src/helpers/metrics.ts

```diff
@@ -0,0 +1,33 @@
+import init, { client } from '@snapshot-labs/snapshot-metrics';
+import db from './mysql';
+import type { Express } from 'express';
+
+export default function initMetrics(app: Express) {
+  init(app, { whitelistedPath: [/^\/$/] });
+}
+
+new client.Gauge({
+  name: 'subscribers_per_status_count',
+  help: 'Number of subscribers per status',
+  labelNames: ['status'],
+  async collect() {
+    [
+      ['active', '`key` IS NOT NULL'],
+      ['pending', '`key` IS NULL']
+    ].forEach(async function callback(this: any, data: any) {
+      this.set(
+        { status: data[0] },
+        (await db.queryAsync(`SELECT count(*) as count FROM \`keys\` WHERE ${data[1]}`))[0]
+          .count as any
+      );
+    }, this);
+  }
+});
```

Comment:
Much cleaner this way
```suggestion
export default function initMetrics(app: Express) {
  init(app, { whitelistedPath: [/^\/$/] });
}

async function collectSubscriberCounts() {
  const subscriberCounts = await Promise.all([
    db.queryAsync('SELECT count(*) as count FROM `keys` WHERE `key` IS NOT NULL'),
    db.queryAsync('SELECT count(*) as count FROM `keys` WHERE `key` IS NULL')
  ]);

  return [
    { status: 'active', count: subscriberCounts[0][0].count },
    { status: 'pending', count: subscriberCounts[1][0].count }
  ];
}

new client.Gauge({
  name: 'snapshot_subscriber_counts',
  help: 'Number of Snapshot subscribers by status',
  labelNames: ['status'],
  async collect() {
    const subscriberCounts = await collectSubscriberCounts();

    subscriberCounts.forEach(({ status, count }) => {
      this.set({ status }, count);
    });
  }
});
```